### ISSUE TITLE
`ConfirmationDialog` misses theme overrides section

### ISSUE THREAD

**Issue by MathiasBaumgartinger** [godotengine/godot#86712]
### Tested versions

- Reproducible in 4.1.3, 4.2.1, 4.2

### System information

Manjaro Linux (KDE)

### Issue description

`ConfirmationDialog` does not show a section for theme override. This is really confusing because the user would expect the internal panel of the confirmation dialog to have the color of the set theme which is not the case, as internally, the override is set anyways. Interestingly, an `AcceptDialog` (parent in the inhertiance chain) shows the section in the inspector properly (see screenshots below).

![image](https://github.com/godotengine/godot/assets/33001106/896c4247-2b1f-4508-90b9-d9c7c25b64c4)
![image](https://github.com/godotengine/godot/assets/33001106/6733b14d-2f14-40bd-8ad6-5a678b9dc3c2)

### Steps to reproduce

Create `ConfirmationDialog` in editor and go to inspector, theme overrides will be missing.

The unexpected output can be seen in the remote node hierarchy. Navigate to the ConfirmationDialog and find the internal panel child. It will have a theme override applied: 
![image](https://github.com/godotengine/godot/assets/33001106/10b6375c-94c9-4b5f-aae6-f9e99abaf887)

### Minimal reproduction project (MRP)

Not necessary

**Comment by jsjtxietian** [comment]
Yes it is confusing. I find that add the same setting as `AcceptDialog` for `ConfirmationDialog` in `default_theme.cpp` can make the ui appear again, but I'm not sure whether it's the correct way of fixing this.

**Comment by MathiasBaumgartinger** [comment]
An easy workaround for now:

```gdscript
@export var panel_style: StyleBox

func _ready():
    add_theme_stylebox_override("panel", panel_style)
```
